crates/rpc/lean/src/lib.rs

```diff
@@ -1,14 +1,55 @@
-pub fn add(left: u64, right: u64) -> u64 {
-    left + right
+use std::sync::Arc;
+
+use actix_web::{App, HttpServer, dev::ServerHandle, middleware, web::Data};
+use config::LeanRpcServerConfig;
+use ream_chain_lean::lean_chain::LeanChain;
+use tokio::sync::RwLock;
+use tracing::info;
+
+use crate::routes::register_routers;
+
+pub mod config;
+pub mod handlers;
+pub mod routes;
+
+/// Start the Lean API server.
+pub async fn start_lean_server(
+    server_config: LeanRpcServerConfig,
+    lean_chain: Arc<RwLock<LeanChain>>,
+) -> std::io::Result<()> {
+    info!(
+        "starting HTTP server on {:?}",
+        server_config.http_socket_address
+    );
+    let stop_handle = Data::new(StopHandle::default());
+
+    let server = HttpServer::new(move || {
+        let stop_handle = stop_handle.clone();
+        App::new()
+            .wrap(middleware::Logger::default())
+            .app_data(stop_handle)
+            .app_data(Data::new(lean_chain.clone()))
+            .configure(register_routers)
+    })
+    .bind(server_config.http_socket_address)?
+    .run();
+
+    server.await
+}
+
+#[derive(Default)]
+struct StopHandle {
+    inner: parking_lot::Mutex<Option<ServerHandle>>,
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+#[allow(dead_code)]
+impl StopHandle {
+    pub(crate) fn register(&self, handle: ServerHandle) {
+        *self.inner.lock() = Some(handle);
+    }
```

Comment:
if we aren't using this code can we remove it, that goes for the beacon copy as well, also code shouldn't be duplicated, a 3rd crate should be made to avoid overlaping code. but in this case the StopHandle can just be removed as it isn't used, or if you want the code can be used